activestorage/app/models/active_storage/blob/analyzable.rb

```diff
@@ -42,16 +42,21 @@ def analyzed?
     analyzed
   end
 
+  # Returns true if the blob's analyzer is present.
+  def analyze_later?
+    analyzer_class.present?
+  end
```

Comment:
Another thing we could try:

```ruby
def analyze_later
  if analyzer.analyze_later?
     ActiveStorage::AnalyzeJob.perform_later(self)
  else
    analyzer.analyze
  end
end
```

Reply:
I tried this approach and I think it is better. It gives us the option of enabling inline analysis for other analyzers and the logic is easier to follow.